### PR TITLE
chore: add documentation for uploading security scoped resources

### PR DESCRIPTION
**Pull request by phantumcode**
_Issue #, if available:_

_Description of changes:_
* Update Swift Storage Upload documentation to include steps to access and upload security scoped resources
image

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/fragments/lib/storage/ios/upload.mdx [hunk 1]
```diff
@@ -138,6 +138,48 @@ resultSink = uploadTask
 
 </BlockSwitcher>
 
+### Working with Security Scoped Resources
```
Comment (hdworld11) [inline src/fragments/lib/storage/ios/upload.mdx:141]:
```suggestion
### Working with Security Scoped Resources (from iCloud)
```

**Review by hdworld11 — APPROVED** [review]

**Comment by atierian** [comment]
Linking issue for visibility https://github.com/aws-amplify/docs/issues/4405

**Comment by atierian** [comment]
Can we include links to the Apple docs for the APIs referenced here?
[startAccessingSecurityScopedResource()](https://developer.apple.com/documentation/foundation/url/1779698-startaccessingsecurityscopedreso)
[stopAccessingSecurityScopedResource()](https://developer.apple.com/documentation/foundation/url/1780153-stopaccessingsecurityscopedresou)

**Comment by phantumcode** [comment]
> 
Added links to Apple's documentation.

File: src/fragments/lib/storage/ios/upload.mdx [hunk 2]
```diff
@@ -138,6 +138,48 @@ resultSink = uploadTask
 
 </BlockSwitcher>
 
+### Working with Security Scoped Resources (from iCloud)
+Security scoped resources refer to files that are retrieved from iCloud.
```
Comment (atierian) [inline src/fragments/lib/storage/ios/upload.mdx:142]:
```suggestion
Security scoped resources refer to files that are retrieved from iCloud or other cloud storage providers. You're likely to run into these file types when using system components that provide access to files stored in iCloud, e.g. [UIDocumentBrowserViewController](https://developer.apple.com/documentation/uikit/uidocumentbrowserviewcontroller).

Comment (atierian) [reply src/fragments/lib/storage/ios/upload.mdx:142]: Recommend adding some more context around security scoped resources and when customers may encounter them.

**Review by atierian — CHANGES_REQUESTED** [review]

File: src/fragments/lib/storage/ios/upload.mdx (hunk 2 above)
Comment (phantumcode) [reply src/fragments/lib/storage/ios/upload.mdx]: Updated per recommendation.

File: src/fragments/lib/storage/ios/upload.mdx [hunk 3]
```diff
@@ -138,6 +138,48 @@ resultSink = uploadTask
 
 </BlockSwitcher>
 
+### Working with Security Scoped Resources (from iCloud)
+Security scoped resources refer to files that are retrieved from iCloud. 
+
+To upload security scoped resources:
+1. you'll need to use `startAccessingSecurityScopedResource()` and `stopAccessingSecurityScopedResource()` to access the data within security scoped files
+2. temporarily persist the data from the security scoped files in your app's sandbox 
+3. upload files using the temporary URLs
```
[the remark this answers is not in the document]
Comment (phantumcode) [reply src/fragments/lib/storage/ios/upload.mdx]: Updated per suggestion.

**Review by atierian — APPROVED** [review]
LGTM

**Review by hdworld11 — APPROVED** [review]

**Pull request merged**